index.d.ts

```diff
@@ -26,8 +26,10 @@ export interface RouterOnChangeArgs {
     router: Router;
     url: string;
     previous?: string;
+    previousRoute?: preact.VNode,
```

Comment:
Can you elaborate on how you'd use this? Just curious.

Reply:
This is already what `current` is. This PR just brings current/previous in parity with one-another. Currently `previous` is a string, while `current` is a Route instance. Not only is this inconsistent, it makes it hard / impossible to implement logic that uses props from the previous route. 

I am depending on these changes in a project of mine, where I use this data to determine whether the user is navigating to/from the default route: [use-case](https://github.com/peabnuts123/pet-game/blob/c5333696a076edfc51f81706afaf1a7cf9b76dde/src/www/src/app.tsx#L29).

Reply:
Ignoring the oddness of calling a browser refresh, why watch the props rather than the route? 

Watching component props like this to me sounds like an anti-pattern. I'm not saying that this shouldn't get merged or anything but that there's almost certainly a better way to do what you want.

Reply:
I'd love to find a work around. I need to know information about the previous route, but I cannot resolve this from the URL. I could manually build up a dictionary of this data myself, keyed by URL, and try and do this lookup myself (including mapping URLs to dictionary keys i.e. trimming query params, etc.), but really it is the responsibility of the Router to give me this information (especially, in my use-case above, whether a Route is the default route or not).